ctfcli/cli/challenges.py

```diff
@@ -51,27 +51,35 @@ def templates(self):
 
         Templates().list()
 
-    def add(self, repo):
+    def add(self, repo, category=None):
         config = load_config()
 
         if repo.endswith(".git"):
             # Get relative path from project root to current directory
             challenge_path = Path(os.path.relpath(os.getcwd(), get_project_path()))
 
-            # Get new directory that will exist after clone
+            # Get new directory that will add the git subtree
             base_repo_path = Path(os.path.basename(repo).rsplit(".", maxsplit=1)[0])
 
-            # Join targets
-            challenge_path = challenge_path / base_repo_path
+            # Set the challenge path's category directory
+            if category:
+                category_path = Path(category)
+                # Join targets
+                challenge_path = challenge_path / category_path / base_repo_path
+            else:
+                challenge_path = challenge_path / base_repo_path
             print(challenge_path)
 
             config["challenges"][str(challenge_path)] = repo
 
+            # Setup remote for a user to easily push and pull git subtree
+            subprocess.call(["git", "remote", "add", base_repo_path, repo])
+            print(f"Added git remote {base_repo_path}")
+            subprocess.call(["git", "subtree", "add", "--prefix", challenge_path, base_repo_path, "master", "--squash"])
+
```

Comment:
Defaulting to master is likely going to break at somepoint. Can it just default to the origin default branch? I think adding the remote may be a mistake but I'm not sure. Can it be done instead with perhaps something like `ctf challenge push`?

Reply:
Also note the issue with the master branches in the later lines.